### PR TITLE
 Suggest `next` when trying to break from captured block 

### PR DESCRIPTION
**Pull request by r00ster91**
When doing this:
```cr
def method(&block)
  block.call
end

method do
  break
end
```
then you get `Error in line 6: can't break from captured block`.
Now this makes the error message suggest `next` because when using `break` in a captured block you most likely tried to escape from the captured block and `next` makes that possible.
Also when you instead try to `return` from a captured block, it gives you a helpful error message: `Error in line 6: can't return from captured block, use next` so why not suggest `next` in case of `break` too?

This also downcases `next`'s and `break`'s error message just like all the other error messages.

### CODE REVIEW

**Comment by straight-shoota** [comment]
In general, this is probably a good feature. But I'm not sure if "use next" is the best advice. It's not always as simple as that, because `next` and `break`/`return` have different semantics. If the block is called multiple times, `next` might not give you what you want. So I think it should be a more relaxed suggestion.

**Comment by asterite** [comment]
@straight-shoota Right on! I was thinking about `did you mean 'next'`? But that doesn't give much info either.

In general, I think we should improve all of our error messages to look a bit more like Elm, where they take a lot of time to explain what went wrong and suggestions with their explanations.

**Comment by r00ster91** [comment]
What about sometimes linking to the docs?
```
can't break from captured block.
If you want to exit the captured block, use `next`, however `next` will have behave differently when used in a block or a loop inside of the captured block.
For more information see https://crystal-lang.org/reference/syntax_and_semantics/next.html.
```

**Comment by Sija** [comment]
Ruby allows `break` within blocks which makes sense, since `next` provide different semantics and breaking from the block is a side-effect.

**Comment by Sija** [comment]
Actually, why it's not possible to use `break` in captured blocks?

**Comment by asterite** [comment]
@Sija `break` in non-captured blocks means "exit the enclosing method". For a capture block it can't work.

```crystal
def capture(&block)
  block
end

def foo
  # This break looks like it will exit foo, but that can't happen
  block = capture { break }
  puts 1
  block
end

block = foo
block.call
```

**Comment by r00ster91** [comment]
I hope this is clear enough now. I don't think it needs to be as long as I proposed. In the end it might rather confuse the user.

**Comment by straight-shoota** [comment]
Indeed, that is way to long and too much detail. A simple "Maybe you can use `next` instead?`" would suffice.

File: spec/compiler/semantic/block_spec.cr [hunk 1]
```diff
@@ -1075,7 +1075,7 @@ describe "Block inference" do
 
       bar
       ),
-      "can't break from captured block, use `next`, however `next` will have behave differently when used in a block or a loop inside of the captured block."
+      "can't break from captured block, try using `next` to exit the block."
```
Comment (straight-shoota) [inline spec/compiler/semantic/block_spec.cr]: I think stating that `next` exits the block is misleading exactly in the way we're trying to clarify things.

**Comment by Sija** [comment]
@asterite Now it makes sense, thanks for explanation!

**Pull request merged**